apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/PathExpectationSamples.kt

```diff
@@ -29,12 +28,362 @@ class PathExpectationSamples {
+        dir.newFile("$TEST_FILE.txt")
         fails {
             expect(dir).toBeAnEmptyDirectory()
         }
     }
+
+    @Test
+    fun toStartWith() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir.last()).toStartWith(Paths.get(TEST_DIR))
+
+        fails {
+            expect(dir.last()).toStartWith(Paths.get(INVALID_DIR))
+        }
+    }
+
+    @Test
+    fun notToStartWith() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir.last()).notToStartWith(Paths.get(INVALID_DIR))
+
+        fails {
+            expect(dir.last()).notToStartWith(Paths.get(TEST_DIR))
+        }
+    }
+
+    @Test
+    fun toEndWith() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).toEndWith(Paths.get(TEST_DIR))
+
+        fails {
+            expect(dir).toEndWith(Paths.get(INVALID_DIR))
+        }
+    }
+
+    @Test
+    fun notToEndWith() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).notToEndWith(Paths.get(INVALID_DIR))
+
+        fails {
+            expect(dir).notToEndWith(Paths.get(TEST_DIR))
+        }
+    }
+
+    @Test
+    fun toExist() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).toExist()
+
+        fails {
+            expect(Paths.get(INVALID_DIR)).toExist()
+        }
+    }
+
+    @Test
+    fun notToExist() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(Paths.get(INVALID_DIR)).notToExist()
+
+        fails {
+            expect(dir).notToExist()
+        }
+
+    }
+
+    @Test
+    fun toBeReadable() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).toBeReadable()
+
+        fails {
+            expect(Paths.get(INVALID_DIR)).toBeReadable()
+        }
+    }
+
+    @Test
+    fun toBeWritable() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).toBeWritable()
+
+        fails {
+            expect(Paths.get(INVALID_DIR)).toBeWritable()
+        }
+    }
+
+    @Test
+    fun toBeExecutable() {
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).toBeExecutable()
+
+        fails {
+            expect(Paths.get(INVALID_DIR)).toBeExecutable()
+        }
+    }
+
+    @Test
+    fun toBeARegularFile() {
+        val file = tempDir.newFile(TEST_FILE)
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(file).toBeARegularFile()
+
+        fails {
+            expect(dir).toBeARegularFile()
+        }
+    }
+
+
+    @Test
+    fun toBeADirectory() {
+        val file = tempDir.newFile(TEST_FILE)
+        val dir = tempDir.newDirectory(TEST_DIR)
+
+        expect(dir).toBeADirectory()
+
+        fails {
+            expect(file).toBeADirectory()
+        }
+    }
+
+    @Test
+    fun toBeAbsolute() {
+        val dir = tempDir.newDirectory(ABSOLUTE_DIR)
+
+        expect(dir).toBeAbsolute()
+
+        fails {
+            expect(Paths.get(RELATIVE_DIR)).toBeAbsolute()
+        }
```

Comment:
I would not create a directory here as it is not necessary (also ABSOLUTE_DIR is not per se absolute, it's the newDirectory which returns an absolute path and the user might not know this). Better rewrite this to the following which makes it clearer:

```suggestion
        expect(Paths.get("/absolute/path").toBeAbsolute()

        fails {
            expect(Paths.get("relative/path")).toBeAbsolute()
        }
```
same same for toBeRelative

Reply:
`Paths.get("/absolute/path").isAbsolute` returns false.
We could use `FileSystemView.getFileSystemView().homeDirectory.toPath().resolve("foo")` instead and maybe even refactor it as method inside of test class or somewhere else.
`private fun getAbsolutePath(name: String) = FileSystemView.getFileSystemView().homeDirectory.toPath().resolve(name)`

Reply:
I guess you are using Windows. On a unix system it returns true. I figured this was abstracted away by Java in nio but apparently not (learned something new :slightly_smiling_face:). In this case you need to write it as follows:
```kotlin
    // on unix like systems / and \ on Windows
    val s = FileSystems.getDefault().getSeparator()
    expect(Paths.get("${s}absolute${s}path").toBeAbsolute()

    fails {
        expect(Paths.get("relative${s}path")).toBeAbsolute()
    }
```

Reply:
on windows it still returns false, but "\\absolute\path" works fine.

Reply:
I investigated further and utility class WindowsPathParser decides in this case that "\absolute\path" is relative.

Reply:
That's a pity, so probably we ned to prefix the path with `C:` in case of Windows. Can you quickly check if this would work?

Reply:
C:\absolute\path returns true

Reply:
Will this work?
![Screenshot_43](https://user-images.githubusercontent.com/39156859/120636285-83238a80-c47e-11eb-8a9a-adca17d77565.png)

Reply:
I guess it will but it is somehow also a bit too much for a simple example (which should also be suitable for beginners).
I therefore suggest we change it to the following which takes up less space so the most relative thing:
```
val s = FileSystems.getDefault().separator
val prefix = if (s == "\\") "C:" else ""
expect(Paths.get("$prefix${s}absolute${s}path")).toBeAbsolute()
```
What do you think?

Reply:
yup I understand your point, I will go with your solution